ang/test/civicase/case/actions/services/webforms-case-action.service.spec.js

```diff
@@ -23,24 +25,102 @@
-          it('displays the action link', function () {
-            expect(WebformsCaseAction.isActionAllowed(webformAction, cases, attributes)).toBeTrue();
+        describe('webforms list in case actions dropdown', () => {
+          beforeEach(function () {
+            attributes.mode = 'case-details';
+          });
+
+          describe('when webforms are present', () => {
+            it('displays the action link', function () {
+              expect(WebformsCaseAction.isActionAllowed(webformAction, cases, attributes)).toBeTrue();
+            });
+          });
+
+          describe('when webforms are not present', () => {
+            beforeEach(function () {
+              webformAction.items = [];
+            });
+
+            it('hides the action link', function () {
+              expect(WebformsCaseAction.isActionAllowed(webformAction, cases, attributes)).toBeFalse();
+            });
           });
         });
 
-        describe('when webforms are not present', () => {
+        describe('webforms list in case summary header', () => {
           beforeEach(function () {
-            webformAction.items = [];
+            attributes.mode = 'case-details-header';
+          });
+
+          describe('when webforms are present', () => {
+            it('hides the action link', function () {
+              expect(WebformsCaseAction.isActionAllowed(webformAction, cases, attributes)).toBeFalse();
+            });
           });
 
-          it('displays the action link', function () {
-            expect(WebformsCaseAction.isActionAllowed(webformAction, cases, attributes)).toBeFalse();
+          describe('when webforms are not present', () => {
+            beforeEach(function () {
+              webformAction.items = [];
+            });
+
+            it('hides the action link', function () {
+              expect(WebformsCaseAction.isActionAllowed(webformAction, cases, attributes)).toBeFalse();
+            });
+          });
+        });
+      });
+
+      describe('when webforms should be visible in a separate list', function () {
+        beforeEach(function () {
+          webformsList.isVisible = true;
+        });
+
+        describe('webforms list in case actions dropdown', () => {
+          beforeEach(function () {
+            attributes.mode = 'case-details';
+          });
+
+          describe('when webforms are present', () => {
+            it('hides the action link', function () {
+              expect(WebformsCaseAction.isActionAllowed(webformAction, cases, attributes)).toBeFalse();
+            });
+          });
+
+          describe('when webforms are not present', () => {
+            beforeEach(function () {
+              webformAction.items = [];
+            });
+
+            it('hides the action link', function () {
+              expect(WebformsCaseAction.isActionAllowed(webformAction, cases, attributes)).toBeFalse();
+            });
+          });
+        });
```

Comment:
we are also missing the visible scenario here.

Reply:
"when webforms should be visible in a separate list", "webforms list in case actions dropdown" should not be visible, so there is not visible scenario in this case.